private/src/avb_watchdog/IasSystemdWatchdogManager.cpp

```diff
@@ -0,0 +1,334 @@
+
+    uint64_t const nextTimeoutTimestamp = IasSystemdWatchdogManager::getCurrentRawTime() + currentResetTimespan;
+    mWatchdogTimer.setTimeoutInterval(currentResetTimespan);
+
+    IasWatchdogResult const setTimerResult = mWatchdogTimerRegistration->registerTimer(mWatchdogTimer);
+
+    if (setTimerResult.toString() == IasMediaTransportAvb::IasResult::cOk.toString())
+    //if ( IAS_FAILED(setTimerResult))
+    {
+      DLT_LOG_CXX(mDltContext, DLT_LOG_ERROR, "IasSystemdWatchdogManager failed to set very high priority watchdog timer with timeout value (ms)", currentResetTimespan, setTimerResult);
+    }
+    else
+    {
+      mNextTimeoutTimestamp = nextTimeoutTimestamp;
+    }
+  }
+}
+
+void IasSystemdWatchdogManager::checkResetWatchdog()
+{
+  if (   ( mResetTimespan > 0LL )
+      && (IasSystemdWatchdogManager::getCurrentRawTime() > mNextTimeoutTimestamp) )
+  {
+    DLT_LOG_CXX(mDltContext, DLT_LOG_INFO, "IasSystemdWatchdogManager::checkResetWatchdog trigger timeout explicitly");
+    notifyTimedOut();
+  }
+}
+
+void IasSystemdWatchdogManager::forceResetWatchdog()
+{
+  if ( mResetTimespan > 0LL )
+  {
+    DLT_LOG_CXX(mDltContext, DLT_LOG_VERBOSE, "IasSystemdWatchdogManager::forceResetWatchdog trigger systemd watchdog");
+    notifyTimedOut();
+  }
+}
+
+IasWatchdogResult IasSystemdWatchdogManager::init(std::shared_ptr<IasWatchdogTimerRegistration> watchdogTimerRegistration)
+{
+  if(watchdogTimerRegistration == nullptr)
+  {
+    return IasMediaTransportAvb::IasResult::cParameterInvalid;
+  }
+
+  mWatchdogTimerRegistration = watchdogTimerRegistration;
+
+  IasWatchdogResult result = IasMediaTransportAvb::IasResult::cOk;
+
+  char const * const envStr = getenv("WATCHDOG_USEC");
+  if (envStr != nullptr)
+  {
+    uint64_t intValue = 0;
+    std::stringstream stream;
+    stream << envStr;
+    stream >> intValue;
+    if ( stream.bad() )
+    {
+      DLT_LOG_CXX(mDltContext, DLT_LOG_ERROR, "IasSystemdWatchdogManager failed to convert WATCHDOG_USEC environment variable to integer value", envStr);
+      result = IasMediaTransportAvb::IasResult::cInitFailed;
+    }
+    else if (intValue < minimumWatchdogTimeoutValueUSec)
+    {
+      DLT_LOG_CXX(mDltContext, DLT_LOG_ERROR, "IasSystemdWatchdogManager environment variable WATCHDOG_USEC too small. Values below 100000 (which are 100ms) are not acceptable.", envStr, intValue);
+      result = IasMediaTransportAvb::IasResult::cInitFailed;
+    }
+    else
+    {
+      mResetTimespan = uint64_t( intValue/watchdogUSec2TimespanDivisor );
+    }
+  }
+  else
+  {
+    DLT_LOG_CXX(mDltContext, DLT_LOG_WARN, "IasSystemdWatchdogManager no watchdog configuration found in env 'WATCHDOG_USEC'. Disable checking.");
+  }
+
+  if ( result.toString() == IasMediaTransportAvb::IasResult::cOk.toString() && ( mResetTimespan > 0LL) )  //watchdog result type not compatible with IAS_SUCCEED
+  //if ( IasMediaTransportAvb::IAS_SUCCEEDED(result) && ( mResetTimespan > 0LL) )
```

Comment:
I noticed there are a few of these commented statements. Do they come from the original patch? If yes, I think they need to be removed in a later patch.

Reply:
Yes, they're all from the original patch. I believe that's true for all of AVB streamhandler code. If we get into all the cleanup that remains to be done, then it'll probably take a month to merge this. I have verified the basic functionality for this feature at this point, so, I agree that we should merge this now and going forward, we can cleanup/fix the coding style and functional issues as we come across them.